scripts/qa_crowbarsetup.sh

```diff
@@ -1414,47 +1456,73 @@ function enable_ssl_for_nova_dashboard()
     proposal_set_value nova_dashboard default "['attributes']['nova_dashboard']['ssl_no_verify']" true
 }
 
-function hacloud_configure_cluster_defaults()
+function hacloud_configure_cluster_members()
 {
-    clustertype=$1
+    local operation=$1
+    local clustername=$2
     shift
+    shift
+    # default is to add nodes
+    local op="+="
+    if [[ $operation == "set" ]] ; then
+        op="="
+    fi
 
-    nodes=`printf "\"%s\"," $@`
+    local nodes=`printf "\"%s\"," $@`
     nodes="[ ${nodes%,} ]"
-    proposal_set_value pacemaker "$clustertype" \
-        "['deployment']['pacemaker']['elements']['pacemaker-cluster-member']" "$nodes"
+    proposal_modify_value pacemaker "$clustername" \
+        "['deployment']['pacemaker']['elements']['pacemaker-cluster-member']" "[]" "||="
+    proposal_modify_value pacemaker "$clustername" \
+        "['deployment']['pacemaker']['elements']['pacemaker-cluster-member']" "$nodes" "$op"
 
     if [[ "configuration" = "with per_node" ]] ; then
         for node in $@; do
-            proposal_set_value pacemaker "$clustertype" \
+            proposal_set_value pacemaker "$clustername" \
                 "['attributes']['pacemaker']['stonith']['per_node']['nodes']['$node']" "{}"
-            proposal_set_value pacemaker "$clustertype" \
+            proposal_set_value pacemaker "$clustername" \
                 "['attributes']['pacemaker']['stonith']['per_node']['nodes']['$node']['params']" "''"
         done
     fi
+}
+
+function hacloud_configure_cluster_defaults()
+{
+    local clustername=$1
+    local clustertype=$2
+    local cnodes=
 
-    proposal_set_value pacemaker "$clustertype" \
+    # assigning nodes
```

Comment:
here as well being a bit more specific in the comment which nodes are assigned to what wouldn't hurt IMO